File: src/main/scala/com/exasol/cloudetl/helper/JavaClassTypes.scala
package com.exasol.cloudetl.helper

/**
 * An extendable trait that holds Java class types.
 *
 * These class values are used in pattern match to obtain the
 * corresponding values from Exasol iterator.
 */
trait JavaClassTypes {
  val jInteger: Class[java.lang.Integer] = classOf[java.lang.Integer]
  val jLong: Class[java.lang.Long] = classOf[java.lang.Long]
  val jBigDecimal: Class[java.math.BigDecimal] = classOf[java.math.BigDecimal]
  val jDouble: Class[java.lang.Double] = classOf[java.lang.Double]
  val jBoolean: Class[java.lang.Boolean] = classOf[java.lang.Boolean]
  val jString: Class[java.lang.String] = classOf[java.lang.String]
  val jSqlDate: Class[java.sql.Date] = classOf[java.sql.Date]
  val jSqlTimestamp: Class[java.sql.Timestamp] = classOf[java.sql.Timestamp]
}

File: src/main/java/com/exasol/cloudetl/helper/JavaClassTypes.java
package com.exasol.cloudetl.helper;

/** Java class constants used for type dispatch. */
class JavaClassTypes {
    private JavaClassTypes() {
        // not instantiable
    }

    /** Integer class. */
    static final Class<Integer> J_INTEGER = Integer.class;
    /** Long class. */
    static final Class<Long> J_LONG = Long.class;
    /** BigDecimal class. */
    static final Class<java.math.BigDecimal> J_BIG_DECIMAL = java.math.BigDecimal.class;
    /** Double class. */
    static final Class<Double> J_DOUBLE = Double.class;
    /** Boolean class. */
    static final Class<Boolean> J_BOOLEAN = Boolean.class;
    /** String class. */
    static final Class<String> J_STRING = String.class;
    /** SQL date class. */
    static final Class<java.sql.Date> J_SQL_DATE = java.sql.Date.class;
    /** SQL timestamp class. */
    static final Class<java.sql.Timestamp> J_SQL_TIMESTAMP = java.sql.Timestamp.class;
}
